pkg/operator/sync.go

```diff
@@ -167,39 +188,38 @@ func (optr *Operator) syncAll(syncFuncs []syncFunc) error {
 			}
 			break
 		}
-		if err := optr.clearDegradedStatus(sf.name); err != nil {
-			return fmt.Errorf("error clearing degraded status: %w", err)
-		}
-	}
 
-	if err := optr.syncDegradedStatus(syncErr); err != nil {
-		return fmt.Errorf("error syncing degraded status: %w", err)
+		optr.clearDegradedStatus(updatedCO, sf.name)
```

Comment:
Looking at this again this function seems to be just syncDegradedStatus if currently degraded. I'm just wondering if this is technically needed since if syncErr is empty I think the syncDegradedStatus right below would clear it, if we care to go through all the syncs first.

Or maybe this was here for a reason :thinking:

Reply:
I think this was historically here to clear a degrade(from a prior syncAll run) based on the success of a syncFunc, as syncDegradedStatus did a direct call before. But since this is no longer the case(we are batching updates), it might be redundant as `syncDegradedStatus` can clear degrades on its own(assuming we exist the loop with a nil syncError). I'll take this out and see if we break anything